packages/stateManager/mod.ts

```diff
@@ -92,6 +95,26 @@ async function storeOrdersByStatus(
   }
   return store.put(status, orders);
 }
+// Synchronize db operations to prevent race conditions.
+const lockAndRelease = <T extends ShopObjectTypes>(
+  store: Store<T>,
+): Store<T> => {
+  const dbMutex = new Mutex();
```

Comment:
i fear this is not comprehensive. The mutex as to be shared by all stores. There has to be one mutex which governs all gets/puts.